server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -201,9 +205,14 @@ public Float parse(XContentParser parser, boolean coerce) throws IOException {
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, Boolean hasDocValues) {
```

Comment:
There is no need to use `Boolean` here, please  use `boolean`:

```suggestion
            public Query termQuery(String field, Object value, boolean hasDocValues) {
```